src/Hal/Application/Extension/ExtensionService.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * (c) Jean-François Lépine <https://twitter.com/Halleck45>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Hal\Application\Extension;
+
+use Hal\Application\Config\Configuration;
+use Hal\Component\Bounds\Bounds;
+use Hal\Component\Result\ResultCollection;
+
+class ExtensionService {
+
+    /**
+     * @var Repository
+     */
+   private $repository;
+
+    /**
+     * ExtensionService constructor.
+     * @param Repository $repository
+     */
+    public function __construct(Repository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @return Repository
+     */
+    public function getRepository()
```

Comment:
Every time this method is called the call looks like `$extensionService->getRepository()->all()`. We might consider replacing `getRepository()` with a `getAllExtensions()` method.